src/api/app/models/package.rb

```diff
@@ -1516,10 +1519,10 @@ def update_activity
     # the value we add to the activity, when the object gets updated
     addon = 10 * (Time.now.to_f - updated_at_was.to_f) / 86400
     addon = 10 if addon > 10
-    new_activity = activity + addon
-    new_activity > 100 ? 100 : new_activity
 
-    self.activity_index = new_activity
+    # atomar update to avoid dead locks
+    sql = ApplicationRecord.connection
+    sql.execute("UPDATE packages SET activity_index = LEAST(100, #{Package._activity_algorithm} + #{addon})  WHERE id=#{id}")
```

Comment:
And why would that fix the deadlock?

Reply:
it is now a single SQL command instead of two depending